libraries/helpers.rb

```diff
@@ -124,7 +124,7 @@ def splunk_runas_user
     end
 
     def splunk_service_provider
-      if node['init_package'] == 'systemd'
+      if systemd?
```

Comment:
This whole method should go away as we are no longer supporting any distribution that isn't using systemd.

Reply:
However, even the systems that support systemd can still be running SysV scripts, depending on the targeted deployment envrionment. Splunk installs need to be able to know what environment they're going in. This method needs to stay.

Reply:
Removing this method in this PR is also way out of scope for this PR. This PR is submitted to address a specific BUG, not to completely rewrite the cookbook -- the latter of which would be a breaking change. Removing the helper method would be a breaking change, not a bugfix.

Reply:
True on being out of scope. The Chef 17 fixes really should happen first in a separate PR and then this PR rebased against that once it's done. That PR could deal with this issue specfically.

Reply:
Until that happens, this PR is ready to go, IMHO.